website/source/api/secret/totp/index.html.md

```diff
@@ -0,0 +1,272 @@
+
+This documentation assumes the TOTP backend is mounted at the
+`/totp` path in Vault. Since it is possible to mount secret backends at
+any location, please update your API calls accordingly.
+
+## Create Key
+
+This endpoint creates or updates a key definition.
+
+| Method   | Path                         | Produces                                                                  |
+| :------- | :--------------------------- | :------------------------------------------------------------------------ |
+| `POST`   | `/totp/keys/:name`           | if generating a key: `200 application/json` else: `204 (empty body)`      |
+
+### Parameters
+
+- `name` `(string: <required>)` – Specifies the name of the key to create. This is specified as part of the URL.
+
+- `generate` `(bool: false)` – Specifies if a key is generated by Vault or if a key is being passed from another service.
+
+- `key_size` `(int: 20)` – Specifies the size in bytes of the Vault generated key. Only used if generate is true.
+
+- `url` `(string: "")` – Specifies the TOTP key url string that can be used to configure a key. Only used if generate is false.
+
+- `key` `(string: <required - if generate is false and url is empty>)` – Specifies the master key used to generate a TOTP code. Only used if generate is false.
+
+- `issuer` `(string: "" <required - if generate is true and url is empty>)` – Specifies the name of the key’s issuing organization.
+
+- `account_name` `(string: "" <required - if generate is true and url is empty>)` – Specifies the name of the account associated with the key.
+
+- `period` `(int or duration format string: 30)` – Specifies the length of time in seconds used to generate a counter for the TOTP code calculation.
+
+- `algorithm` `(string: "SHA1")` – Specifies the hashing algorithm used to generate the TOTP code. Options include "SHA1", "SHA256" and "SHA512".
+
+- `digits` `(int: 6)` – Specifies the number of digits in the generated TOTP code. This value can be set to 6 or 8.
+
+- `skew` `(int: 0)` – Specifies the number of delay periods that are allowed when validating a TOTP code. This value can be either 0 or 1. Only used if generate is true.
+
+- `qr_size` `(int: 200)` – Specifies the pixel size of the square QR code when generating a new key. Only used if generate is true.
+
+### Sample Payload
+
+```json
+{
+  "url": "otpauth://totp/Google:test@gmail.com?secret=Y64VEVMBTSXCYIWRSHRNDZW62MPGVU2G&issuer=Google"
+}
+```
+
+### Sample Request
+
+```
+$ curl \
+    --header "X-Vault-Token: ..." \
+    --request POST \
+    --data @payload.json \
+    https://vault.rocks/v1/totp/keys/my-key
+```
+
+### Sample Payload
+
+```json
+{
+  "generate": true,
+  "issuer": "Google",
+  "account_name": "test@gmail.com",
+}
+```
+
+### Sample Request
+
+```
+$ curl \
+    --header "X-Vault-Token: ..." \
+    --request POST \
+    --data @payload.json \
+    https://vault.rocks/v1/totp/keys/my-key
+```
+
+### Sample Response
```

Comment:
Please add some documentation here indicating that the returned QR code, if any, is a base64-encoded PNG.